server/src/main/java/org/elasticsearch/bootstrap/ESPolicy.java

```diff
@@ -157,16 +158,17 @@ public boolean implies(ProtectionDomain domain, Permission permission) {
     }
 
     @SuppressForbidden(reason = "We get given an URL by the security infrastructure")
-    private boolean canAccessSecuredFile(URL location, FilePermission permission) {
-        if (location == null) {
+    private boolean canAccessSecuredFile(ProtectionDomain domain, FilePermission permission) {
+        if (domain == null) {
             return false;
         }
+        URL location = domain.getCodeSource().getLocation();
 
         // check the source
         Set<URL> accessibleSources = securedFiles.get(permission);
         if (accessibleSources != null) {
             // simple case - single-file referenced directly
-            return accessibleSources.contains(location);
+            return accessibleSources.contains(location) || system.implies(domain, new AllPermission());
```

Comment:
This deserves some explanation, can you add a comment? Can we also create a static AllPermission so every access does not require an object creation?

Reply:
I'll certainly do so if this is what we end up doing, but I admit I assumed this would also be untenable as a long-term solution. We do check to prevent AllPermission grants otherwise; is that sufficient to allow this to be safe in line with the original intent? If you think it has legs, I should probably do another pass to make sure everything is in order.

Reply:
It depends what the hit is, but I don't suspect the extra check would be a problem. Secured files are only config files, so they shouldn't be on any hot path. It isn't the cleanest, but I also don't think we should dwell on this too long given the imminent removal of security manager.